src/responder/common/cache_req/plugins/cache_req_group_by_id.c

```diff
@@ -64,6 +64,12 @@ cache_req_group_by_id_lookup(TALLOC_CTX *mem_ctx,
                              struct sss_domain_info *domain,
                              struct ldb_result **_result)
 {
+   if ((domain->id_max != 0 && data->id > domain->id_max)
+        ||  (data->id < domain->id_min)) {
+        DEBUG(SSSDBG_OP_FAILURE, "gid exceeds min/max boundaries\n");
+        return EOK;
```

Comment:
Yes, you need to return some special error code here. Additionally, please check both id_max and id_min for being non-zero, like this:
```
 if ((domain->id_min && (data->id < domain->id_min)) ||
     (domain->id_max && (data->id > domain->id_max))) {
```

Reply:
Also, please make the debug message level FUNC_DATA or something as low as that so that the debug message doesn't appear during normal operation

Reply:
Finally, please add a case handler into the switch statement in cache_req_search_cache() so that we only print a very low-level debug message and don't hit the "default" handler.